File: frontend/app/models/address.ts
import { Model } from 'app/models/model';
import { Suburb } from 'app/models/suburb';

export class Address extends Model {

	public latitude: number;
	public longitude: number;
	public house_number: string;
	public place: string;
	public postal_code: string;
	public street: string;
	public suburb_id: string;
	public suburb: Suburb;

	constructor(json: any = {} as Address) {
		super(json.id);
		this.latitude = json.latitude && json.latitude || 0;
		this.longitude = json.longitude && json.longitude || 0;
		this.house_number = json.house_number && json.house_number || '0';
		this.place = json.place && json.place || '';
		this.postal_code = json.postal_code && json.postal_code || '';
		this.street = json.street && json.street || '';
		this.suburb_id = json.suburb_id && json.suburb_id || '';
		this.suburb = json.suburb && new Suburb(json.suburb) || new Suburb();
	}

	get toString(): string {
		return this.isValid()
			? (
				this.street + ' ' +
				this.house_number + ' ' +
				this.postal_code + ' ' +
				this.place + ' ' +
				(this.suburb ? this.suburb.name : ''))
			: '';
	}

	public compareTo(address: Address): boolean {
		return address.street.toLocaleLowerCase().localeCompare(this.street.toLocaleLowerCase()) === 0 &&
			address.house_number.toLocaleLowerCase().localeCompare(this.house_number.toLocaleLowerCase()) === 0 &&
			address.postal_code.toLocaleLowerCase().localeCompare(this.postal_code.toLocaleLowerCase()) === 0 &&
			address.place.toLocaleLowerCase().localeCompare(this.place.toLocaleLowerCase()) === 0;
	}

	public isValid(): boolean {
		return this.place &&
			this.postal_code &&
			this.street &&
			this.house_number !== '0';
	}

}

File: server/wupportal/src/main/java/de/codeschluss/wupportal/model/Address.java
package de.codeschluss.wupportal.model;

import java.io.Serializable;
import java.util.List;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonIgnore;

import de.codeschluss.wupportal.activity.ActivityEntity;
import de.codeschluss.wupportal.base.BaseEntity;
import de.codeschluss.wupportal.organisation.OrganisationEntity;

/**
 * The persistent class for the addresses database table.
 * 
 */
@Entity
@Table(name = "addresses")
public class Address extends BaseEntity implements Serializable {
	
	private static final long serialVersionUID = 1L;

	@Column(name = "house_number")
	private String houseNumber;

	private String place;

	@Column(name = "postal_code")
	private String postalCode;

	private String street;

	@ManyToOne
	@JsonIgnore
	private Suburb suburb;

	private float latitude;

	private float longitude;

	@OneToMany(mappedBy = "address")
	@JsonIgnore
	private List<ActivityEntity> activities;

	@OneToMany(mappedBy = "address")
	@JsonIgnore
	private List<OrganisationEntity> organisations;

	public Address() {
		super();
	}

	public String getHouseNumber() {
		return this.houseNumber;
	}

	public void setHouseNumber(String houseNumber) {
		this.houseNumber = houseNumber;
	}

	public float getLatitude() {
		return this.latitude;
	}

	public void setLatitude(float latitude) {
		this.latitude = latitude;
	}

	public float getLongitude() {
		return this.longitude;
	}

	public void setLongitude(float longitude) {
		this.longitude = longitude;
	}

	public String getPlace() {
		return this.place;
	}

	public void setPlace(String place) {
		this.place = place;
	}

	public String getPostalCode() {
		return this.postalCode;
	}

	public void setPostalCode(String postalCode) {
		this.postalCode = postalCode;
	}

	public String getStreet() {
		return this.street;
	}

	public void setStreet(String street) {
		this.street = street;
	}

	public List<ActivityEntity> getActivities() {
		return this.activities;
	}

	public void setActivities(List<ActivityEntity> activityEntities) {
		this.activities = activityEntities;
	}

	public Suburb getSuburb() {
		return this.suburb;
	}

	public void setSuburb(Suburb suburb) {
		this.suburb = suburb;
	}

	public List<OrganisationEntity> getOrganisations() {
		return this.organisations;
	}

	public void setOrganisations(List<OrganisationEntity> organisationEntities) {
		this.organisations = organisationEntities;
	}

}